Run the apps using docker compose

NithinKuruba opened the pull request:
Fixes https://trello.com/c/wDVjmuOa

junminahn reviewed (commented):
I have three concerns:
1. does the `localhost` DB URL is available in other docker containers? I believe `network_mode: "host"` is that for?
2. how are we going to pass secret environments such as `keycloak client secrets`?
- maybe this example helps? https://github.com/bcgov/cognito_example_client_app/blob/main/examples/next-app-jwt/docker-compose.yml
3. how do we sync the codebase within the docker containers in the local dev environment?

NithinKuruba commented:
> I have three concerns:
> 
> 1. does the `localhost` DB URL is available in other docker containers? I believe `network_mode: "host"` is that for?
> 2. how are we going to pass secret environments such as `keycloak client secrets`?
> 
> * maybe this example helps? https://github.com/bcgov/cognito_example_client_app/blob/main/examples/next-app-jwt/docker-compose.yml
> 
> 3. how do we sync the codebase within the docker containers in the local dev environment?

@junminahn yes, the network_mode enables services to run in host network and are accessible through `localhost`

I have updated the dockerfile and docker compose to keep the code base sync between host and the docker containers for ui and backend apps. 

Regarding the secrets, you can export them (Ex.: export CLIENT_SECRET=xxx) and pass them as env vars to docker compose services. I have added an example (not used by app) to our `sso-portal-ui` service that reads  `CLIENT_SECRET` from the exported secret.
